src/components/Menu/Menu.tsx

```diff
@@ -0,0 +1,119 @@
+import { Menu as HeadlessMenu } from '@headlessui/react';
+import clsx from 'clsx';
+import React from 'react';
+
+import styles from './Menu.module.css';
+import type { ExtractProps } from '../../util/utility-types';
+
+import ClickableStyle from '../ClickableStyle';
+import Icon from '../Icon';
+import type { IconName } from '../Icon';
+import PopoverContainer from '../PopoverContainer';
+import PopoverListItem from '../PopoverListItem';
+
+// Note: added className here to prevent private interface collision within HeadlessUI
+export type MenuProps = ExtractProps<typeof HeadlessMenu> & {
+  /**
+   * Allow custom classes to be applied to the menu container.
+   */
+  className?: string;
+};
+
+export type MenuItemProps = ExtractProps<typeof HeadlessMenu.Item> & {
+  href?: string;
+  icon?: IconName;
+  onClick?: () => any;
+};
+
+export type MenuButtonProps = ExtractProps<typeof HeadlessMenu.Button>;
+export type MenuItemsProps = ExtractProps<typeof HeadlessMenu.Items>;
+
+/**
+ * BETA: This component is still a work in progress and is subject to change.
+ *
+ * `import {Menu} from "@chanzuckerberg/eds";`
+ *
+ * Creates a list of actions which appear next to a trigger component. This should
+ * be used when there is a discrete number of actions such as user navigations (e.g.,
+ * a profile menu with links to settings) or a set of actions
+ */
+export const Menu = (props: MenuProps) => {
```

Comment:
In our xfn meeting we landed on using Menu/Select for the two components to avoid ambiguity. All three of these things sorta drop down, and we already have other (deprecated) components using "Dropdown" in the name, so trying to avoid that in the component itself.

The figma currently has conflicting naming too, so will take a note for that to get tidied up

Reply:
Thanks!